src/approaches/nsrt_rl_approach.py

```diff
@@ -72,10 +90,128 @@ def learn_from_interaction_results(
                                       actions,
                                       _is_demo=False,
                                       _train_task_idx=i)
-            self._online_trajectories.append(traj)
+            self._train_task_to_online_traj[i] = traj
+
+
+        option_to_data = {} # option_name -> online experience data
+
+        # For each task, for each _Option involved in the trajectory, compute
+        # and store (s, a, s', r, relative_param) data.
+        for i in range(len(self._train_tasks)):
+            plan = self._train_task_to_option_plan[i]
+            traj = self._train_task_to_online_traj[i]
+
+            curr_option_idx = 0
+            curr_option = plan[curr_option_idx]
+            if curr_option.name not in option_to_data:
+                option_to_data[curr_option.name] = []
+            curr_states = [traj.states[0]]
+            curr_actions = []
+            curr_rewards = []
+            curr_relative_params = []
+            actions = (a for a in traj.actions)
+            block = [b for b in curr_option.objects if b.type.name=='block'][0]
+            robot = [r for r in curr_option.objects if r.type.name=='robot'][0]
+
+            # Generate transition data, (s, a, s', r, relative_param). The
+            # reward R(s, a, s') = neg_reward if s' is not within epsilon of the
+            # subgoal, and pos_reward otherwise.
+            for j, s in enumerate(traj.states[1:]):
+                curr_states.append(s)
+                curr_actions.append(next(actions))
+                # TODO: is there a better way to do this based on
+                # curr_option.parent._changing_obj_feats? Couldn't access this
+                # field, but could access curr_option.parent._changing_var_order
+                if curr_option.params[-1] > 0:  # if holding becomes true (pick)
+                    changing_obj_feats = np.array([
+                        s.get(block, 'grasp'),
+                        s.get(robot, 'x'),
+                        s.get(robot, 'y'),
+                        s.get(robot, 'grip'),
+                        s.get(robot, 'holding')
+                    ])
+                else:
+                    changing_obj_feats = np.array([
+                        s.get(block, 'x'),
+                        s.get(block, 'grasp'),
+                        s.get(robot, 'x'),
+                        s.get(robot, 'grip'),
+                        s.get(robot, 'holding')
+                    ])
+                curr_relative_params.append(curr_option.memory["absolute_params"] - changing_obj_feats)
+
+                if curr_option.terminal(s):
+                    # TODO: make this not hardcoded to be environment specific,
+                    # or implement specifically per environment and throw an
```

Comment:
definitely we will want to do all of this in a non environment specific way

Reply:
Solution to earlier comment about line 122 solves it here as well.